DESIGN.md

```diff
@@ -0,0 +1,37 @@
+# Apollo Server design goals
+
+The goal of this project is to build a GraphQL server for Node.js that is simple, flexible, and performant. It is a Node.js GraphQL server built for production use.
```

Comment:
@nnance If there are issues in graphql-js that we need to be fixed right away, I think we would fork graphql-js and use our fork in apollo server until the PR lands in graphql-js. If possible I'd like to avoid maintaining a fork forever. If there is no way around maintaining a fork, it should be a community effort for a production-optimized version of graphql in js, and not something only apollo server uses.